### PR TITLE
Use `markdown-cli` to check markdown for the style errors

### PR DESCRIPTION
**Pull request by mathbunnyru**

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (mathbunnyru) [inline .pre-commit-config.yaml:31]: This fixes some errors in an automatic way.

File: .markdownlint.yaml
Comment (mathbunnyru) [inline .markdownlint.yaml:7]: This is better to be fixed in a separate PR (it will be too much noise here)

**Comment by consideRatio** [comment]
This LGTM!

I have become very fond of an autoformatter for markdown called prettier, but I have no experience with markdown-cli to compare it with. Due to that, I suggest that deliberating on that isn't a blocker for merging this PR. Briefly though, we have used prettier across the jupyterhub organization for YAML, JSON, and Markdown autoformatting - [here is an example](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/d45e83131ee8e9920171fe7b031e28ca53f89cd7/.pre-commit-config.yaml#L35-L39) from a .pre-commit-config.yml:

```yaml
  # Autoformat: markdown, yaml
  - repo: https://github.com/pre-commit/mirrors-prettier
    rev: v2.2.1
    hooks:
      - id: prettier
```

**Comment by mathbunnyru** [comment]
> This LGTM!
> 
> I have become very fond of an autoformatter for markdown called prettier, but I have no experience with markdown-cli to compare it with. Due to that, I suggest that deliberating on that isn't a blocker for merging this PR. Briefly though, we have used prettier across the jupyterhub organization for YAML, JSON, and Markdown autoformatting - [here is an example](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/d45e83131ee8e9920171fe7b031e28ca53f89cd7/.pre-commit-config.yaml#L35-L39) from a .pre-commit-config.yml:
> 
> ```yaml
>   # Autoformat: markdown, yaml
>   - repo: https://github.com/pre-commit/mirrors-prettier
>     rev: v2.2.1
>     hooks:
>       - id: prettier
> ```

Thanks, @consideRatio. Prettier looks nice. I will take a look more closely it after this PR gets merged.

**Comment by mathbunnyru** [comment]
@romainx could you please take a look here?

Also, I had one approval from @consideRatio, which makes me believe that this is ready to be merged :)

**Comment by mathbunnyru** [comment]
The reason I'm asking - this blocks me from further docs improvement (from the technical point of view).

**Comment by romainx** [comment]
@mathbunnyru there is a conflict now could you solve it?
Any advice on how to review all the changes to the docs?
I guess generate the doc locally and check?

**Comment by consideRatio** [comment]
> Any advice on how to review all the changes to the docs?
> I guess generate the doc locally and check?

RTD pr preview would be nice to activate, requires maintainer permissions on the associated RTD project though.

**Comment by mathbunnyru** [comment]
> @mathbunnyru there is a conflict now could you solve it?

Fixed.

> Any advice on how to review all the changes to the docs?
> I guess generate the doc locally and check?

Yes, for now local generation (though we need to use old versions).
Or, just check that the syntax is correct, it should work fine :)

RTD pr preview would be great 👍

**Comment by mathbunnyru** [comment]
> @mathbunnyru and @romainx I don't remember how it works properly, but perhaps you can make sure you have a RTD account (or have signed in once using your github handle to RTD)?

I've already created one and linked it with my GitHub account and the login is the same `mathbunnyru`.
So, waiting for someone to give me some rights :)

**Comment by romainx** [comment]
@consideRatio same for me.

**Review by romainx — APPROVED** [review]
Finally I reviewed the files manually 😅. Did not catch any error 🕵️ .

**Comment by romainx** [comment]
Thanks @mathbunnyru always improve !

**Pull request merged**
